internal/provisioner/plain/pkg/git/git.go

```diff
@@ -0,0 +1,78 @@
+package git
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/operator-framework/rukpak/api/v1alpha1"
+)
+
+const (
+	defaultDirectory = "./manifests"
+	defaultBranch    = "main"
+)
+
+type CheckoutCmd struct {
+	v1alpha1.GitSource
+}
```

Comment:
Did you consider a single exported function (e.g. `CloneCommandFor(v1alpha1.GitSource) (string, error)` that combines the validation and command building into a single entrypoint with no need for an extra struct?

Reply:
I think that approach also works, I had something like that at one point before moving to this struct-based implementation.

Reply:
I think my only hesitation is that we can easily index something like a tag/commit/directory/etc. with a custom type, while reduce the coupling with an individual bundle type? It's also entirely possible that we don't have any need for this custom type implementation right now, and all we're interested in functionality centered around building up a formatted strings that's responsible for sourcing git bundle contents.

Reply:
I decided to combine both implementations: return a (string, error) in the main codepath and use an internal type under the hood, which helps with building the correct command.